classes/class.ilMatrixChatPlugin.php

```diff
@@ -112,10 +112,12 @@ public function jsFolder(string $file = ""): string
 
     public function getUsernameSchemeVariables(): array
     {
+	$truncLength = $this->getPluginConfig()->getExternalUserTruncateLength();
+	
         return [
             "CLIENT_ID" => CLIENT_ID,
-            "LOGIN" => $this->user->getLogin(),
-            "EXTERNAL_ACCOUNT" => $this->user->getExternalAccount()
+            "LOGIN" => substr($this->user->getLogin(),0,-$truncLength),
+            "EXTERNAL_ACCOUNT" => substr($this->user->getExternalAccount(),0,-$truncLength)
```

Comment:
@mglaubitz / @mBeym: `substr` is not multi-byte safe. If we really want to shorten a string based on the length of the number of characters (and not bytes), we should use `ilStr::subStr(...)`.